### ISSUE TITLE
write a simple prediction py script

### ISSUE THREAD

**Issue by jianhuaixie** [affinelayer/pix2pix-tensorflow#103]
the pix2pix.py script is so complex,Do you write a simple python script for the predition function? by the way ,you could make a server for prediction.

**Comment by caiostringari** [comment]
A predict mode would be very helpful indeed!

**Comment by nidetaoge** [comment]
def expand_dim4array(src_image):
    image = src_image.copy()
    image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
    image = cv2.resize(image, (int(round(a.scale_size*a.aspect_ratio)), a.scale_size))  # width, height

    image = np.asarray(image, dtype=np.float32)
    image = image.reshape((image.shape[0], image.shape[1], 3))
    image = np.expand_dims(image, axis=0)

    return image


def TestWithModelForSingleImage():
    if tf.__version__.split('.')[0] != "1":
        raise Exception("Tensorflow version 1 required")

    if not os.path.exists(a.output_dir):
        os.makedirs(a.output_dir)
        
    if a.checkpoint is None:
            raise Exception("checkpoint required for test mode")    

    input_paths = glob.glob(os.path.join(a.input_dir, "*.jpg"))
    if len(input_paths) == 0:
        input_paths = glob.glob(os.path.join(a.input_dir, "*.png"))


    def get_name(path):
        name, _ = os.path.splitext(os.path.basename(path))
        return name

    # if the image names are numbers, sort by the value rather than asciibetically
    # having sorted inputs means that the outputs are sorted in test mode
    
    ### input_dir
    if all(get_name(path).isdigit() for path in input_paths):
        input_paths = sorted(input_paths, key=lambda path: int(get_name(path)))
    else:
        input_paths = sorted(input_paths)


    # inputs and targets are [batch_size, height, width, channels]
    image_placeholder = tf.placeholder(tf.float32, (1, a.scale_size, int(round(a.scale_size*a.aspect_ratio)), 3), name="Images")

    input_tensor = preprocess(image_placeholder)	
	
    outputs = create_model(input_tensor)
    outputs = deprocess(outputs)

    def convert(image):
        if a.aspect_ratio != 1.0:
            # upscale to correct aspect ratio
            size = [a.scale_size, int(round(a.scale_size * a.aspect_ratio))]
            image = tf.image.resize_images(image, size=size, method=tf.image.ResizeMethod.BICUBIC)		

            #return tf.image.convert_image_dtype(image, dtype=tf.uint8, saturate=True)
            return tf.image.convert_image_dtype(image, dtype=tf.float64)
			
    # reverse any processing on images so they can be written to disk or displayed to user
    converted_outputs = convert(outputs)

    float_converted_outputs = tf.to_float(converted_outputs, name='final_results')

    saver = tf.train.Saver(max_to_keep=1)

    logdir = a.output_dir
    sv = tf.train.Supervisor(logdir=logdir, save_summaries_secs=0, saver=None)
    with sv.managed_session() as sess:

        if a.checkpoint is not None:
            print("loading model from checkpoint")
            checkpoint = tf.train.latest_checkpoint(a.checkpoint)
            saver.restore(sess, checkpoint)
	
	for op in tf.get_default_graph().get_operations():
	    print(op.name)

        image_dir = os.path.join(a.output_dir, "images")
        if not os.path.exists(image_dir):
            os.makedirs(image_dir)
        for step in range(len(input_paths)):

            src_image = cv2.imread(os.path.join(a.input_dir, basename(input_paths[step])), 1)
            image = expand_dim4array(src_image)
			
            feed_dict = {image_placeholder: image}
	    out_path = os.path.join(image_dir, basename(input_paths[step]))

            result = sess.run(float_converted_outputs, feed_dict=feed_dict)

            result = result * 255.0
            dst_image = np.array(result[0]).astype(np.uint8)
            dst_image = cv2.cvtColor(dst_image, cv2.COLOR_RGB2BGR)

            cv2.imwrite(out_path, dst_image)  
            print("evaluated image", input_paths[step])


This is my prediction script to use pretrained model to work on image. You may need to make some changes to work for your job.

**Comment by chenyuZha** [comment]
@nidetaoge Thanks for share your script. I have some questions for the code: 

- `for op in tf.get_default_graph().get_operations():
    print(op.name)` may not need to be here since you don't use any node for feed your tensor.

- Can your explain a little why here you use `float_converted_outputs` ?

- for the function `create_model`, as it has just one input( instead of 2), so I guess you just take the part that create generator from this function ?

I tested your script but fail to have a good result as `pix2pix.py`, thanks advance for all of the explications!

**Comment by arshdefoamed** [comment]
@nidetaoge 
Hi, 
I tried using your script and it has too many compilation issues. Can you upload the complete the script that you are using?

**Comment by Supagoat** [comment]
I just generate a test set with a blank target and run it in --mode test

**Comment by purijs** [comment]
> I just generate a test set with a blank target and run it in --mode test

Wont this convert your input image into blank?

**Comment by RicFraga** [comment]
I have this simple script to create outputs, hope it helps you all

```
def create_output(model, input):
  img = tf.cast(tf.image.decode_jpeg(tf.io.read_file(input)), tf.float32)
  img = tf.image.resize(img, [256, 256])
  img = img / 127.5 - 1
  prediction = model(img[np.newaxis, ...], training = True) * 0.5 + 0.5

  return prediction[0, ...]
```
And to use it you just pass the generator model and the path of the input image

```
a = create_output(generator, '/content/galleta.jpeg')

plt.imshow(a)
plt.axis('off')
plt.show()
```
